pyteal/ast/router.py

```diff
@@ -715,12 +999,31 @@ def build_program(self) -> tuple[Expr, Expr, sdk_abi.Contract]:
             * clear_state_program: an AST for clear-state program
             * contract: a Python SDK Contract object to allow clients to make off-chain calls
         """
+        if not self.bare_call_actions.is_empty():
+            bare_call_approval = self.bare_call_actions.approval_construction()
+            if bare_call_approval:
+                self.approval_ast.bare_calls = [
+                    CondNode(
+                        Txn.application_args.length() == Int(0),
+                        cast(Expr, bare_call_approval),
+                    )
+                ]
+
+        optimize = optimize if optimize else OptimizeOptions()
```

Comment:
```suggestion
        optimize = optimize or OptimizeOptions()
```

ignoreable suggestion

Reply:
yes, nice move